src/deferred.js

```diff
@@ -359,5 +363,22 @@ jQuery.extend( {
 	}
 } );
 
+if ( window.console && window.console.warn ) {
+
+	// These usually indicate a programmer mistake during development,
+	// warn about them ASAP rather than swallowing them by default.
+
+	var rerrorNames = /^(Eval|Internal|Range|Reference|Syntax|Type|URI)Error$/;
+
+	jQuery.Deferred.exceptionHook = function( error ) {
+		if ( error && rerrorNames.test( error.name ) ) {
+			window.console.warn( "jQuery.Deferred exception: " + error.message );
```

Comment:
There's a guard above for that case but you're right, it needs to be in the function itself to prevent a really nasty Heisenbug where it would crash, but only if you didn't have Dev Tools open. :smiling_imp: 

In environments without the console my intent was that you just wouldn't get any output, which is basically no worse than the current default. I think escalating to something that is still there like `window.alert` would be too intrusive, but someone could add that as a hook if they were trying to debug an IE9-specific problem.